cds/modules/deposit/tasks.py

```diff
@@ -122,3 +133,28 @@ def preserve_celery_states_on_db():
     db.session.commit()
     if ids:
         RecordIndexer().bulk_index(iter(ids))
+
+
+@shared_task(ignore_result=True, rate_limit="100/m")
+def index_deposit_projects(check_time=None):
+    cache = current_cache.get("task_index_deposit_projects:details") or {}
+    if "update_date" not in cache:
+        # Set the update date by default to 10 minutes ago
+        cache["update_date"] = check_time or (
+            datetime.utcnow() - timedelta(minutes=10)
+        )
+    records = RecordMetadata.query.filter(
+        RecordMetadata.updated > cache["update_date"]
+    ).with_entities(RecordMetadata.id, RecordMetadata.json)
+
+    projects_ids = []
+    for _id, json in records:
+        if json and json.get("$schema") == current_jsonschemas.path_to_url(
+            Project._schema
+        ):
+            projects_ids.append(str(_id))
+
+    if projects_ids:
+        RecordIndexer().bulk_index(iter(projects_ids))
+
+    current_cache.set("task_index_deposit_projects:details", cache, timeout=-1)
```

Comment:
Shouldn't the update_date be set to now() at the end of the task?
The import for `current_cache` is missing